### PR TITLE
Cleanups to README, CHANGELOG, CONTRIBUTING

### PR DESCRIPTION
**Pull request by jamesonwilliams**
You can see the rendered copies of these change to the following files:
1. [README.md](https://github.com/aws-amplify/amplify-android/pull/413/files?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8)
2. [CHANGELOG.md](https://github.com/aws-amplify/amplify-android/pull/413/files?short_path=4ac32a7#diff-4ac32a78649ca5bdd8e0ba38b7006a1e)
3. [CONTRIBUTING.md](https://github.com/aws-amplify/amplify-android/pull/413/files?short_path=6a33714#diff-6a3371457528722a734f3c51d9238c13)

-----------------

Simplifies the flow of information in the `README.md`. Removes marketing
language "so easy!" etc. Simplifies the project description.
    
Moves development information into the `CONTRIBUTING.md`, and re-orients
the project `README.md` towards end-users, not developers.
    
Removes the `CHANGELOG.md`, the presence of which had dangerously misled
the reader to believe it portrayed an accurate and complete depiction of
pre-release changes.  A change log will either be provided via the
GitHub releases page, or will be resumed in a markdown document in this
repository for the 1.0 release.

Fixes the index in the `CONTRIBUTING.md`, which had not included many of
the sections of the document. The new index includes a section that was
moved from `README.md`, _"Consuming Development Versions of the
Framework"_.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,88 +1,130 @@
...
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-}
-```
+## Report a Bug
 
-### Using AWS Mobile Client
+We apprecaite your feedback -- comments, questions, and bug reports. Please
```
Comment (richardmcclellan) [inline README.md:119]:
```suggestion
We appreciate your feedback -- comments, questions, and bug reports. Please
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,13 +55,50 @@ Load this project into Android Studio by selecting File > Open, and choosing
...
+
+To publish the outputs of the build, execute the following command from
+the root of the `amplify-android` project:
+
+```
+./gradlew publishToMavenLocal
+```
+
+After this, you can use the published development artifacts from an app.
+To do so, specify `mavenlocal()` inside the app's top-level
```
Comment (richardmcclellan) [inline CONTRIBUTING.md:82]:
```suggestion
To do so, specify `mavenLocal()` inside the app's top-level
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -43,13 +55,50 @@ Load this project into Android Studio by selecting File > Open, and choosing
...
+`build.gradle` file:
+
+```gradle
+buildscript {
+    repositories {
+        mavenlocal() // this should ideally appear before other repositories
```
Comment (richardmcclellan) [inline CONTRIBUTING.md:88]:
```suggestion
        mavenLocal() // this should ideally appear before other repositories
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -43,13 +55,50 @@ Load this project into Android Studio by selecting File > Open, and choosing
...
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.6.3'
+    }
+}
+
+allprojects {
+    repositories {
+        mavenlocal() // this should ideally appear before other repositories
```
Comment (richardmcclellan) [inline CONTRIBUTING.md:97]:
```suggestion
        mavenLocal() // this should ideally appear before other repositories
```

**Comment by jamesonwilliams** [comment]
> Minor suggestions, overall LGTM!

Thanks for your catches, @richardmcclellan! Applied all.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -12,11 +12,23 @@ contribution.
...
+    + [Build and Validate Your Work](#build-and-validate-your-work)
+    + [Run Instrumentation Tests](#run-instrumentation-tests)
   * [Reporting Bugs/Feature Requests](#reporting-bugs-feature-requests)
   * [Contributing via Pull Requests](#contributing-via-pull-requests)
+  * [Troubleshooting](#troubleshooting)
+    + [Environment Debugging](#environment-debugging)
+    + [Problems with the Build](#problems-with-the-build)
+    + [Failing Instrumentation Tests](#failing-instrumentation-tests)
+  * [Related Repositories](#related-repositories)
+  * [Finding contributions to work on](#finding-contributions-to-work-on)
```
Comment (jpignata) [inline CONTRIBUTING.md:28]: Nit: The rest of the list Uses Title Case, but the new items only capitalize the first letter. I don't have a strong opinion on which is better, but would advise consistency.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -12,11 +12,23 @@ contribution.
...
     + [Adding code to support a new feature](#adding-code-to-support-a-new-feature)
+    + [Build and Validate Your Work](#build-and-validate-your-work)
+    + [Run Instrumentation Tests](#run-instrumentation-tests)
   * [Reporting Bugs/Feature Requests](#reporting-bugs-feature-requests)
   * [Contributing via Pull Requests](#contributing-via-pull-requests)
+  * [Troubleshooting](#troubleshooting)
+    + [Environment Debugging](#environment-debugging)
+    + [Problems with the Build](#problems-with-the-build)
+    + [Failing Instrumentation Tests](#failing-instrumentation-tests)
+  * [Related Repositories](#related-repositories)
```
Comment (jpignata) [inline CONTRIBUTING.md:27]: Idle thought: I'm surprised the CLI is not on this list given it is more pertinent to an Android application developer than the other libraries.

File: README.md [hunk 7]
```diff
@@ -1,88 +1,129 @@
...
 
-AWS Amplify provides a high-level interface to perform different categories of cloud operations. Each category can be implemented by a _plugin_. Our default plugins support interaction with Amazon Web Services (AWS), but AWS Amplify is designed to be extensible to any other backend or service.
+AWS Amplify provides a high-level interface to perform different
+categories of cloud operations. Each category is fulfilled by a
+_plugin_. You specify which plugins to use during setup.
+
+The default plugins that we provide are designed to facilitate
+interaction with Amazon Web Services (AWS). But, the Amplify framework
+is designed to be extensible to any other backend or service -- even a
+competing cloud provider!
```
Comment (jpignata) [inline README.md:12]: I might strike the em-dash part of the second sentence.

File: README.md [hunk 8]
```diff
@@ -1,88 +1,129 @@
...
+
+### [REST APIs](https://docs.amplify.aws/lib/restapi/getting-started/q/platform/android):
+
+Easy auth and request signing against multiple REST endpoints. Our
+default plugin works great with Amazon API Gateway.
+
+### [GraphQL APIs](https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/android)
+
+Data modeling and simple auth against GraphQL endpoints. Our default
+plugin targets Amazon AppSync.
```
Comment (jpignata) [inline README.md:34]: Nit: AWS AppSync, but since you mentioned it above, just "AppSync" will suffice.

File: README.md [hunk 9]
```diff
@@ -1,88 +1,129 @@
...
+If you're new to the project, checkout the
+[Getting Started Guide](https://docs.amplify.aws/lib/getting-started/setup/q/platform/android).
+
+## The Categories
+
+### [DataStore](https://docs.amplify.aws/lib/datastore/getting-started/q/platform/android):
+
+Model your app's data. Save, query, and observe changes to your data
+from a local repository. Let DataStore synchronize your local data with
+the Cloud. Our default implemenation syncs local data to/from a DynamoDB
```
Comment (jpignata) [inline README.md:23]: The first mention of a service should be its fully qualified name (i.e. Amazon DynamoDB), and any subsequent mention can use its short name (i.e. DynamoDB).

File: README.md [hunk 10]
```diff
@@ -1,88 +1,129 @@
...
+Collect and report usage data for your app. Our default plugin
+communicates with Amazon Pinpoint.
 
-- [**Analytics**](https://docs.amplify.aws/lib/analytics/getting-started?platform=android): Easily collect and report usage data for your app.
-- [**API**](https://docs.amplify.aws/lib/graphqlapi/getting-started?platform=android): An interface for interacting with REST and GraphQL endpoints. Provides AWS Signature Version 4 signing.
-- [**DataStore**](https://docs.amplify.aws/lib/datastore/getting-started?platform=android): Provides a persistent on-device storage repository for you to write, read, and observe changes to data. Data is sycnhronized to the cloud as well as across devices.
-- [**Storage**](https://docs.amplify.aws/lib/storage/getting-started?platform=android): Manage user content for your app, storing it in public, protected or private buckets.
+### [Storage](https://docs.amplify.aws/lib/storage/getting-started/q/platform/android):
+
+Store and retrieve files in the Cloud. We use S3 by default.
```
Comment (jpignata) [inline README.md:43]: First mention: Amazon Simple Storage Service (Amazon S3), subsequent mentions: Amazon S3.

File: README.md [hunk 11]
```diff
@@ -1,88 +1,129 @@
...
-    // Specify only the modules that the app will use
+    // Only specify modules that provide functionality your app will use
     implementation 'com.amplifyframework:aws-datastore:0.10.0'
     implementation 'com.amplifyframework:aws-api:0.10.0'
     implementation 'com.amplifyframework:aws-storage-s3:0.10.0'
     implementation 'com.amplifyframework:aws-analytics-pinpoint:0.10.0'
 }
 ```
 
-### Local Publishing of Artifacts
```
Comment (jpignata) [inline README.md:35]: :thumbsup: This definitely is better situated in the CONTRIBUTING guide.

File: README.md [hunk 12]
```diff
@@ -1,88 +1,129 @@
...
+### AWS Mobile Client
 
-``` gradle
-buildscript {
-    repositories {
-        mavenLocal() // This should ideally appear before other repositories
-    }
-    dependencies {
-        classpath 'com.android.tools.build:gradle:3.6.1'
+The default plugins for Amplify Android rely on the AWS Mobile Client to
```
Comment (jpignata) [inline README.md:82]: AWS Mobile Client is a somewhat drab, generic name whose Google results are confusing. Perhaps we can link to something for someone to read up? Our SDK auth guide?

File: README.md [hunk 13]
```diff
@@ -1,88 +1,129 @@
...
-You need to ensure `AWSMobileClient` is [initialized](https://docs.amplify.aws/lib/auth/getting-started?platform=android#initialization) before you initalize and configure `Amplify`. Please note that AWSMobileClient initialization is not required when using API or DataStore category with _API Key_ as the authorization mode.
+## Contribute to the Project
 
-## License
+Please see the [Contributor's Guide](./CONTRIBUTING.md).
 
-This library is licensed under the Apache 2.0 License. 
+--------------------------------
 
+[amplify.aws](https://amplify.aws)
```
Comment (jpignata) [inline README.md:129]: Idle thought: This link just floating here at the bottom might not be necessary. I just changed the repo's website to this URL, which I think is more visible.

File: CONTRIBUTING.md (hunk 5 above)
Comment (jamesonwilliams) [reply CONTRIBUTING.md:28]: Will do!

File: CONTRIBUTING.md (hunk 6 above)
Comment (jamesonwilliams) [reply CONTRIBUTING.md:27]: Very true. Adding.

**Comment by jpignata** [comment]
Sorry for the late feedback! Feel free to ignore. Snoozed, lost.

**Comment by jamesonwilliams** [comment]
> Sorry for the late feedback! Feel free to ignore. Snoozed, lost.

Applied here! https://github.com/aws-amplify/amplify-android/pull/415/files

**Pull request merged**